migrations/20250318101300-migrate-attachment-id-field-to-aid.js

```diff
@@ -0,0 +1,169 @@
+module.exports = {
+  async up(db, client) {
+    await db.collection("Attachment").updateMany(
+      {
+        $or: [
+          { datasetId: { $exists: true } },
+          { sampleId: { $exists: true } },
+          { proposalId: { $exists: true } },
+          { id: { $exists: true } },
+          { aid: { $exists: false } },
+        ],
+      },
+      [
+        {
+          $set: {
+            relationships: {
+              // Build relationships array by checking ALL IDs
+              $cond: [
+                {
+                  $or: [
+                    { $ifNull: ["$datasetId", false] },
+                    { $ifNull: ["$sampleId", false] },
+                    { $ifNull: ["$proposalId", false] },
+                  ],
+                },
+                {
+                  $concatArrays: [
+                    // Add dataset relationship if exists
+                    {
+                      $cond: [
+                        { $ifNull: ["$datasetId", false] },
+                        [
+                          {
+                            targetIds: ["$datasetId"],
```

Comment:
I would code each relationship as the relationship between one target and the current attachment.
If the attachment is linked to multiple datasets, you should create multiple relationships.